### PR TITLE
Installer for OS X

### PR DESCRIPTION
**Pull request by kevgo**
@hugobho @martinjaime @helenmedrano @mcharawi @arekkas 

implements #138 

This script installs everything that is needed for productive, large-scale software development on a Mac using Exosphere.
* compiler: Xcode
* package_manager: homebrew
* zeromq development headers
* pkg_config
* Node.js
* Exosphere

Documentation will be updated separately. To run the installer:

```
curl https://raw.githubusercontent.com/Originate/exosphere-sdk/kg-installer/bin/install_osx | sh
```

### CODE REVIEW

**Comment by arekkas** [comment]
awesome, I'll check it out this week!

File: bin/install_osx [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+    echo "${green}Node.JS ${bold}$node_js_version${normal}${green} is okay for Exosphere${normal}"
+    node_js_version_ok=true
+  fi
+}
+
+
+function update_node_js {
+  title "UPDATING NODE.JS"
+  echo "${red}Node.JS ${bold}$node_js_version${normal}${red} found.${normal}"
+  echo "Exosphere needs version 4 or above."
```
Comment (dmh43) [inline bin/install_osx]: Is this accurate? I mistakingly used the sdk with Node v5.11 and could not get it to work right until I switched to node v6

Comment (kevgo) [inline bin/install_osx]: Not fully verified (only for half the commands at this point). It is a goal though, since Node4 is the current LTS version.

Comment (dmh43) [inline bin/install_osx]: 👍  Sounds good to me. As addressed in my (now closed PR), this requirement should work fine for the SDK. Carry on!

**Pull request closed without merging**